packages/web-components/src/horizontalbarchart/horizontalbarchart.ts

```diff
@@ -0,0 +1,561 @@
+
+          if (j !== i) {
+            legendRect.style['backgroundColor'] = 'transparent';
+            legendText.style['opacity'] = '0.67';
+          } else {
+            legendRect.style['backgroundColor'] = this.uniqueLegends[j].color!;
+            legendText.style['opacity'] = '1';
+          }
+        }
+      });
+      legendButtonRefs[i].addEventListener('blur', () => {
+        if (this._selectedLegend !== '') {
+          return;
+        }
+
+        for (let j = 0; j < bars.length; j++) {
+          bars[j].style['opacity'] = '1';
+        }
+        for (let j = 0; j < legendButtonRefs.length; j++) {
+          const legendRect = (
+            legendButtonRefs[j].getElementsByClassName('legendRect') as HTMLCollectionOf<HTMLDivElement>
+          )[0];
+          legendRect.style['backgroundColor'] = this.uniqueLegends[j].color!;
+
+          const legendText = (
+            legendButtonRefs[j].getElementsByClassName('legendText') as HTMLCollectionOf<HTMLDivElement>
+          )[0];
+          legendText.style['opacity'] = '1';
+        }
+      });
+      legendButtonRefs[i].addEventListener('click', () => {
+        if (this._selectedLegend === this.uniqueLegends[i].legend) {
+          this._selectedLegend = '';
+
+          for (let j = 0; j < bars.length; j++) {
+            bars[j].style['opacity'] = '1';
+          }
+          for (let j = 0; j < legendButtonRefs.length; j++) {
+            const legendRect = (
+              legendButtonRefs[j].getElementsByClassName('legendRect') as HTMLCollectionOf<HTMLDivElement>
+            )[0];
+            legendRect.style['backgroundColor'] = this.uniqueLegends[j].color!;
+
+            const legendText = (
+              legendButtonRefs[j].getElementsByClassName('legendText') as HTMLCollectionOf<HTMLDivElement>
+            )[0];
+            legendText.style['opacity'] = '1';
+          }
+        } else {
+          this._selectedLegend = this.uniqueLegends[i].legend;
+
+          for (let j = 0; j < bars.length; j++) {
+            if (bars[j].getAttribute('barinfo') !== this.uniqueLegends[i].legend) {
+              bars[j].style['opacity'] = '0.1';
+            } else {
+              bars[j].style['opacity'] = '1';
+            }
+          }
+          for (let j = 0; j < legendButtonRefs.length; j++) {
+            const legendRect = (
+              legendButtonRefs[j].getElementsByClassName('legendRect') as HTMLCollectionOf<HTMLDivElement>
+            )[0];
+            const legendText = (
+              legendButtonRefs[j].getElementsByClassName('legendText') as HTMLCollectionOf<HTMLDivElement>
+            )[0];
+
+            if (j !== i) {
+              legendRect.style['backgroundColor'] = 'transparent';
+              legendText.style['opacity'] = '0.67';
+            } else {
+              legendRect.style['backgroundColor'] = this.uniqueLegends[j].color!;
+              legendText.style['opacity'] = '1';
+            }
+          }
+        }
```

Comment:
I can't help but notice there's a lot of repetition in this series of event listeners. Could this be abstracted out to a single `highlightActive` handler function?

Reply:
@krkshitij is improving this.